db/migrate/20160923143611_add_action_to_content_view_histories.rb

```diff
@@ -0,0 +1,30 @@
+class AddActionToContentViewHistories < ActiveRecord::Migration
+  def up
+    add_column :katello_content_view_histories, :action, :integer, default: 0
+
+    Katello::ContentViewHistory.reset_column_information
+    Katello::ContentViewHistory.find_each do |history|
+      task_label = history.task.try(:label)
+      next unless task_label
+
+      case task_label
+      when "Actions::Katello::ContentViewVersion::Export"
+        history.action = Katello::ContentViewHistory.actions[:export]
+      when "Actions::Katello::ContentView::Publish"
+        history.action = Katello::ContentViewHistory.actions[:publish]
+      when "Actions::Katello::ContentView::Promote"
+        history.action = Katello::ContentViewHistory.actions[:promotion]
+      when "Actions::Katello::ContentView::Remove"
+        history.action = Katello::ContentViewHistory.actions[:removal]
+      else
+        next
+      end
+
+      history.save!
+    end
+  end
+
+  def down
+    remove_column :katello_content_view_histories, :action
+  end
+end
```

Comment:
Currently I just leave them with a blank action but yea, let me delete them. I'll update the validation too.